solr/contrib/ltr/src/test/org/apache/solr/ltr/feature/TestFieldValueFeature.java

```diff
@@ -56,11 +62,13 @@ public void before() throws Exception {
 
     assertU(commit());
 
-    loadFeature("popularity", FieldValueFeature.class.getName(),
-            "{\"field\":\"popularity\"}");
+    for (String field : FIELD_NAMES) {
+      loadFeature(field, FieldValueFeature.class.getName(),
+              "{\"field\":\""+field+"\"}");
 
-    loadModel("popularity-model", LinearModel.class.getName(),
-            new String[] {"popularity"}, "{\"weights\":{\"popularity\":1.0}}");
+      loadModel(field + "-model", LinearModel.class.getName(),
+              new String[] {field}, "{\"weights\":{\""+field+"\":1.0}}");
+    }
```

Comment:
2/2 ... I think the explanation for the mystery is here i.e. the `LinearModel` constructor signature and the interleaving of feature and model loading results in the models having different definitions of what `allFeatures` means. One solution then would be for the test expectations to account for the multiple meanings of `allFeatures` or (my preference probably) for there to be only one `allFeatures` meaning e.g. via the separation of feature and model loading. Or perhaps you can think of a third solution even?

```
for (String field : FIELD_NAMES) {
  loadFeature(field, ...);
}
for (String field : FIELD_NAMES) {
  loadModel(field + "-model", ...);
}
```

Reply:
Thanks for continuing the investigation!
I think the example you posted here should work, although I am still a bit irritated regarding the `extractAllFeatures = true`.

As user I would expect it to always return all the features that were created, no matter if that happened before or after the model. 
But the code makes it explicit, that these information should not be modified (`allFeatures` is an unmodifiableList).
I think that a nicely placed docString could avoid some confusion.

Reply:
Yes, it's a somewhat quirky behaviour. Operationally it could also cause mysteries e.g. when new features are added to an existing store and the (reasonable) expectation is that they will start producing feature values right away when actually that only happens when the existing model(s) are next re-instantiated. +1 to try and reduce such confusion via documentation somehow.